yeonjae/programmers-1/로또의최고순위와최저순위.js

```diff
@@ -0,0 +1,46 @@
+function solution(lottos, win_nums) {
+  const answer = [];
+  let minRank = 0;
+  let maxRank = 0;
+
+  lottos.forEach((lotto) => {
+    if (win_nums.includes(lotto)) {
+      minRank++;
+      maxRank++;
+    }
+    if (lotto === 0) {
+      maxRank++;
+    }
+  });
+
+  checkGrade(maxRank);
+  checkGrade(minRank);
+
+  return answer;
+
+  function checkGrade(num) {
+    switch (num) {
+      case 2:
+        answer.push(5);
+        break;
+      case 3:
+        answer.push(4);
+        break;
+      case 4:
+        answer.push(3);
+        break;
+      case 5:
+        answer.push(2);
+        break;
+      case 6:
+        answer.push(1);
+        break;
+      default:
+        answer.push(6);
+    }
+  }
+}
```

Comment:
```suggestion
function solution(lottos, win_nums) {
  const rank = [6, 6, 5, 4, 3, 2, 1];
  let minRank = 0;
  let maxRank = 0;

  lottos.forEach((lotto) => {
    if (win_nums.includes(lotto)) {
      minRank++;
      maxRank++;
    }
    if (lotto === 0) {
      maxRank++;
    }
  });

  return [rank[maxRank], rank[minRank]];
}
```
- checkGrade를 이용해서 결과를 도출한 풀이는 좋았습니다. 하지만 배열을 이용하는 방법도 존재하고 실제로 보다싶이 더 짧은 코드로 결과를 도출해낼 수 있습니다.
- 아래 코드는 filter를 이용한 풀이입니다
```js
function solution(lottos, win_nums) {
  const rank = [6, 6, 5, 4, 3, 2, 1];
  const winCount = lottos.filter((lotto) => win_nums.includes(lotto)).length;
  const zeroCount = lottos.filter((lotto) => !lotto).length;
  const bestCount = winCount + zeroCount;

  return [rank[bestCount], rank[winCount]];
}
```